### PR TITLE
API: make FloatingArray.astype consistent with IntegerArray/BooleanArray

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by topper-123 — APPROVED** [review]
LGTM.

File: pandas/core/arrays/masked.py [hunk 1]
```diff
@@ -492,10 +491,6 @@ def astype(self, dtype: AstypeArg, copy: bool = True) -> ArrayLike:
             raise ValueError("cannot convert float NaN to bool")
 
         data = self.to_numpy(dtype=dtype, na_value=na_value, copy=copy)
-        if self.dtype.kind == "f":
-            # TODO: make this consistent between IntegerArray/FloatingArray,
-            #  see test_astype_str
-            return astype_nansafe(data, dtype, copy=False)
```
Comment (phofl) [inline pandas/core/arrays/masked.py:498]: Any side effects apart from the changed test?

Comment (jbrockmendel) [reply pandas/core/arrays/masked.py:498]: there shouldnt be.  if this is done from within a Series/DataFrame it'll get cast to object

**Comment by phofl** [comment]
thx @jbrockmendel

**Pull request merged**